### PR TITLE
feat: Revamp product details page and several product fixes and cleanups

### PR DESCRIPTION
**Pull request by sradevski**

### CODE REVIEW

File: packages/admin-next/dashboard/src/lib/client/products.ts [hunk 1]
```diff
@@ -1,24 +1,20 @@
-import {
-  ProductDeleteRes,
-  ProductListRes,
-  ProductRes,
-} from "../../types/api-responses"
+import { ProductDeleteRes, ProductListRes } from "../../types/api-responses"
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/lib/client/products.ts:1]: suggestion: add http response types to `@medusajs/types`

File: packages/admin-next/dashboard/src/v2-routes/products/product-detail/components/product-sales-channel-section/product-sales-channel-section.tsx [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-// import { useAdminSalesChannels } from "medusa-react"
 import { Trans, useTranslation } from "react-i18next"
 import { ActionMenu } from "../../../../../components/common/action-menu"
+import { useSalesChannels } from "../../../../../hooks/api/sales-channels"
 
 type ProductSalesChannelSectionProps = {
   product: Product
 }
 
+// TODO: The fetched sales channel doesn't contain all necessary info
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/v2-routes/products/product-detail/components/product-sales-channel-section/product-sales-channel-section.tsx:12]: What are you missing?

**Review by olivermrbl — APPROVED** [review]

File: packages/admin-next/dashboard/src/v2-routes/products/product-detail/components/product-sales-channel-section/product-sales-channel-section.tsx (hunk 2 above)
Comment (sradevski) [reply packages/admin-next/dashboard/src/v2-routes/products/product-detail/components/product-sales-channel-section/product-sales-channel-section.tsx:12]: @olivermrbl it only has the ID, but we do `*sales_channels` in the selection, I need to see why.

File: packages/admin-next/dashboard/src/lib/client/products.ts (hunk 1 above)
Comment (sradevski) [reply packages/admin-next/dashboard/src/lib/client/products.ts:1]: @olivermrbl I've done a lot of `any` for now, I'll add all types in one go once I get everything working

**Pull request merged**